### PR TITLE
Chore: Move Hilla modules from platform

### PR DESCRIPTION
**Pull request by ZheSun88**
this PR is aming to move all hilla related modules to hilla repo, so that hilla can release separately.

- create related modules
  - hilla (reuse flow.version)
    - generated `lumo.java` and `material.java` based on flow-components
  - hilla-bom
    - include `flow`, `hilla`, `slf4j` and `jna`  
    - include `vaadin-testbench-bom`
    - replace `vaadin-core` `vaadin` with direct dependencies
    - excludes all kits dependencies
    - exclude `vaadin-testbench` and `vaadin-testbench-junit5` platform dependencies
    - exclude `vaadin-dev-bundle` platform dependency
    - exclude `vaadin-core-jandex` and `vaadin-jandex`  
    - exclude `vaadin-dev` (contains vaadin-dev-server, vaadin-dev-bundle)
  - hilla-dev (reuse flow.version for vaadin-dev-server)
  - hilla-react 
    - generated `lumo.java` and `material.java` based on flow-components
    - excluded `junit` dependency, test scoped
  - hilla-spring-boot-starter (reuse flow.version and spring.boot.version)
  - hilla-react-spring-boot-starter (reuse flow.version and spring.boot.version)
- move the script to generate versions.json for hilla and hilla-react (need to be polished later) 
- move the versions.json file from platform (will consider to auto generate this file)

### CODE REVIEW

File: packages/java/hilla-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    <groupId>dev.hilla</groupId>
+    <artifactId>hilla-bom</artifactId>
+    <packaging>pom</packaging>
+    <name>Hilla Platform (Bill of Materials)</name>
+    <description>Hilla Platform (Bill of Materials)</description>
+    <version>2.2-SNAPSHOT</version>
+    <url>https://hilla.dev</url>
+
+    <properties>
+        <slf4j.version>2.0.4</slf4j.version>
```
Comment (ZheSun88) [inline packages/java/hilla-bom/pom.xml:23]: this should not need, as slf4j.version is in root pom

**Comment by ZheSun88** [comment]
NOTE: 
- need to run `./scripts/generateBoms` to generate the needed hilla-versions.json and hilla-react-versions.json
- 
- seems theme dependencies are needed when excuting build-frontend goal
```
Could not execute build-frontend goal: Error occured during goal execution: Lumo dependency needs to be available on the classpath when using a theme name.Please run Maven with the -e switch (or Gradle with the --stacktrace switch), to learn the full stack trace. -> [Help 1]
```

File: versions.json
Comment (manolo) [inline versions.json:1]: I would not maintain this file here, the generateBoms script should download and parse the equivalent from platform to know the entries, and generate it here

**Review by manolo — CHANGES_REQUESTED** [review]

File: versions.json
Comment (ZheSun88) [reply versions.json:1]: updated.

**Review by manolo — APPROVED** [review]

**Review by manolo — APPROVED** [review]

**Pull request merged**